### PR TITLE
 Starting the LSP server locally inside the JVM (second take)

### PR DESCRIPTION
**Pull request by ddekany**
So this time it uses `server/freemarker-languageserver.jar`. See the commit message.

Note that this PR is after the other pending PR. If you merge that PR first, then this PR is only a single commit.

Also, please see my continued suffering with the development environment setup in the `README.md`... would be great if you describe how you do that. Certainly it's not easy for any other newcomer either.

### CODE REVIEW

File: org.eclipse.lsp4e.freemarker/src/org/eclipse/lsp4e/freemarker/FreemarkerStreamConnectionProvider.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/**
+ *  Copyright (c) 2018 Angelo ZERR.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  Contributors:
+ *  Angelo Zerr <angelo.zerr@gmail.com> - initial API and implementation
```
Comment (angelozerr) [inline org.eclipse.lsp4e.freemarker/src/org/eclipse/lsp4e/freemarker/FreemarkerStreamConnectionProvider.java:9]: please add your name

File: org.eclipse.lsp4e.freemarker/src/org/eclipse/lsp4e/freemarker/LocalStreamConnectionProvider.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
+/**
+ *  Copyright (c) 2018 Angelo ZERR.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  Contributors:
+ *  Angelo Zerr <angelo.zerr@gmail.com> - initial API and implementation
```
Comment (angelozerr) [inline org.eclipse.lsp4e.freemarker/src/org/eclipse/lsp4e/freemarker/LocalStreamConnectionProvider.java:9]: please add your name

File: target-platform/target-platform.target [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8" standalone="no"?>
-<?pde version="3.8"?><target name="bluesky" sequenceNumber="7">
+<?pde version="3.8"?><target name="lsp4-freemarker" sequenceNumber="7">
```
Comment (angelozerr) [inline target-platform/target-platform.target]: rename to lsp4e-freemarker

**Comment by angelozerr** [comment]
>Note that this PR is after the other pending PR. If you merge that PR first, then this PR is only a single commit.

Just few comments to change and I will merge it. Please too regenerate the freemarker-server-all-.jar whith last version of source code (NPE should be fixed)

> Also, please see my continued suffering with the development environment setup in the README.md... would be great if you describe how you do that. Certainly it's not easy for any other newcomer either.

Ok I will see that

**Comment by angelozerr** [comment]
@ddekany please update server/freemarker-languageserver.jar too. It have disabled the uses of tolerant parser which must support position update (otherwise you will have NPE when outline will be refreshed).

**Comment by angelozerr** [comment]
Thanks @ddekany !

**Comment by angelozerr** [comment]
@ddekany I think your class LocalStreamConnectionProvider should be hosted in lsp4e project. Is there any chance that you create a gerrit patch for lsp4e eclipse project? If you have not time, are you OK I create this gerrit patch?

**Comment by ddekany** [comment]
@angelozerr Honestly, I don't feel confident to offer a POC like this to lsp4e itself... I did nearly 0 research. Anyway, since you already know your way around there, feel free to do the gerrit patch. At least we get some feedback on what did I miss...

**Comment by angelozerr** [comment]
Ok thanks for your answer. I have created a bugzilla for lsp4e at https://bugs.eclipse.org/bugs/show_bug.cgi?id=533584 and if lsp4e is interested, I will create a gerrit patch with your work (if you are OK)..

**Pull request merged**